gef-ui/server/controllers/about-facility/index.js

```diff
@@ -116,21 +118,56 @@ const validateAboutFacility = async (req, res) => {
           subFieldErrorRefs: dateFieldsInError,
         });
       } else if (coverStartDateIsFullyComplete) {
+        let hasFormattingError = false;
         const now = new Date();
         const threeMonthsFromNow = add(now, { months: 3 });
+        // schema which ensures that coverStartDate year only contains 4 numbers
+        const yearSchema = Joi.string().length(4).pattern(/^[0-9]+$/).required();
```

Comment:
have a look at the SonarCloud messages around the regex pattern

Reply:
done